pkg/storage/kubernetes/client.go

```diff
@@ -80,7 +80,9 @@ func (i *Client) ListIPPools(ctx context.Context) ([]storage.IPPool, error) {
 	logging.Debugf("listing IP pools")
 	ipPoolList := &whereaboutsv1alpha1.IPPoolList{}
 
-	if err := i.client.List(ctx, ipPoolList, &client.ListOptions{}); err != nil {
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, storage.RequestTimeout)
```

Comment:
darn, this is not using the request timeout that is passed as a cmd line parameter. 

I'll push another PR with that now. Hold your horses for a sec; you'll need a new commit here.